python/pyspark/mllib/linalg/__init__.py

```diff
@@ -1296,9 +1296,19 @@ def asML(self):
         return newlinalg.SparseMatrix(self.numRows, self.numCols, self.colPtrs, self.rowIndices,
                                       self.values, self.isTransposed)
 
-    # TODO: More efficient implementation:
     def __eq__(self, other):
-        return np.all(self.toArray() == other.toArray())
+        if ((type(other) is not type(self)) or
```

Comment:
In scala implementation,  equal means mathematically equal regreadless the reprentation of the implementation.